src/shell.js

```diff
@@ -240,9 +240,9 @@ if (ENVIRONMENT_IS_SHELL) {
 
   if (typeof print !== 'undefined') {
     // Prefer to use print/printErr where they exist, as they usually work better.
-    if (typeof console === 'undefined') console = {};
-    console.log = print;
-    console.warn = console.error = typeof printErr !== 'undefined' ? printErr : print;
+    if (typeof console === 'undefined') console = /** @type{!Console} */({});
+    console.log = /** @type{!function(this:Console, ...*): undefined} */ (print);
+    console.warn = console.error = /** @type{!function(this:Console, ...*): undefined} */ (typeof printErr !== 'undefined' ? printErr : print);
   }
```

Comment:
how do annotations like these help in practice?

Reply:
Oh, this avoids warnings I guess?

Reply:
They are to help the type inference, in most cases it is just to mute warnings - on other occassions they can help prevent Closure from making a bad optimization, or to help it do a good one. (examples of both in a later PR)

Reply:
in this case it is just to mute a warning coming from the assignment.